src/App.js

```diff
@@ -1,23 +1,58 @@
 import React, { Component } from 'react';
-import Main from './Main.js'
+
+import Main from './Main.js';
+import Login from './components/Login';
+import Logout from './components/Logout';
+import api from './api';
+
 import 'bootstrap/dist/css/bootstrap.min.css';
 import './App.css';
 
 class App extends Component {
-
   constructor(props) {
     super(props);
     this.state = {
-      user: null
-    }
+      user: null,
+      users: [],
+    };
   }
 
-  render() {
+  async componentDidMount() {
+    const users = await api.getAllUsers();
+    this.setState({ users });
+  }
 
+  onLogin = async (profileObj) => {
+    let users = this.state.users;
+    let user = users.find((u) => u.email === profileObj.email);
+    if (!user) {
+      await api.createUser({
+        name: profileObj.givenName,
+        email: profileObj.email,
+        private: true,
+      });
+
+      users = await api.getAllUsers();
```

Comment:
This seems like a strange reason to reassign. I think it would make more sense to have the declaration of `const users = ...` here, remove line 26 and have line 27 be `this.state.users.find...`

Reply:
I'm setting users equal to the state users at first to reduce unnecessary api calls. Regardless, we need to reassign users to update the list, since this is creating a new user. I guess we could just add the new one instead of making the call, but it seems good to get a fresh list from the database.